docs/explanation/build-plans.rst

```diff
@@ -0,0 +1,56 @@
+Build plans
+===========
+
+A build plan is a list of platforms to build. It can be defined in a project's
+metadata file and filtered with command-line arguments.
+
+Craft-application's default build planner can build multiple platforms.
+
+Consider the following project metadata:
+
+.. code-block:: yaml
+
+  base: ubuntu@24.04
+  platforms:
+    amd64:
+    arm64:
+      build-on: [amd64, arm64]
+      build-for: [arm64]
+
+If the application executes on an ``amd64`` platform, craft-application will
+create the following build plan:
```

Comment:
Is it worth pointing out that internally we actually create the full build plan and then filter it by `build-on`? Maybe in a note block or something?

The reasoning is that as a developer writing an app with craft-application, I would want to know that I shouldn't be worrying about that extra logic from my application (and should instead let craft-application do it). With remote builds, the application may end up wanting to know the full build plan with all build platforms.

Reply:
Good point.  This concept should be more clear now.